client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java

```diff
@@ -193,7 +205,9 @@ public void close() {
     }
 
     public int getActiveChannelCount() {
-        return allChannels.size();
+        List<Channel> activeChannels = allChannels.stream().filter(Channel::isActive).collect(Collectors.toList());
+        log.debug("Channels that are active are {}", activeChannels);
+        return activeChannels.size();
```

Comment:
I have added a unit test that asserts that items are removed allChannels.
https://netty.io/4.0/api/io/netty/channel/group/ChannelGroup.html also indicates that the channel is removed.
> A closed Channel is automatically removed from the collection, so that you don't need to worry about the life cycle of the added Channel